### DISCUSSION TITLE
Question 1: activation function

### DISCUSSION THREAD

**Discussion by mybirth0407** [Boostcamp-AI-Tech-1-15/ddobokki_log#28] [2. DL Basic]
알고 있는 activation function의 종류에 대해 말해주세요, 이게 면접 질문에서 나온다면 지원자의 어떤 부분을 보고 싶은걸까요?

**Comment by ddeokbboki-good** [comment]
sigmoid, relu, tanh, 
딥러닝이 값을 판단하는 원리를 아는지 확인하려고?

**Reply by ddobokki** [reply]
sigmoid, tanh : 최대한 뉴런이 가지는 모양에 비슷하면서도 학습에 가능한(미분 가능한) 함수들입니다. 두 함수 동떨어진 것이 아니라 x를 변형시키며 변환가능합니다.

Relu: 레이어가 깊어지면서 오는 그래디언트 소실을 해결하기 위해 고안된 활성함수 입니다. 0이상의 값은 미분값이 1이 되기 때문에 역전파 과정에서 그래디언트 소실 문제를 해결할 수 있습니다.

Leaky relu: 0 이하의 값에서 relu의 미분값이 0이 되어 학습이 안되는 경우를 해결하기 위해 나온 함수입니다.

elu: 이전 relu기반 함수들이 0 지점에서 미분불가능 한것을 해결하기 위해 나온 함수입니다.

(정확한 답변인지는 모르겠습니다. 그냥 기억나는대로...)

이걸 물어보는건 기존 방식의 한계점과 그 해결과정을 이해하고 있는지 알고 있는지 아는가를 물어보는게 아닐까요? 기존에 그냥 사용하고 있는 기술들이라고 해도 한계점은 있기 마련이고 그것을 알고 해결방안을 제시할수 있는 인재를 찾고 싶어할 것 같습니다.

**Reply by mybirth0407** [reply]
좋습니다! 그리고 최신 논문/동향을 얼마나 파악하고 있는지에 대해서도 물어보는 역할을 합니다.
relu 의 변환인 gelu, selu 등도 충분히 많이 사용하고 있기 때문에, 최신 논문들을 읽으며 이런 부분들도 체크해보시면 좋을 듯 하고,
maxout이라는 속도는 느리지만 가장 성능이 좋은 activation func도 있습니다. 이런 것들을 아는 게 공부 과정에서 중요한지는 저도 여전히 의문이지만, 실무에서는 충분히 활용해볼 수 있기 때문에 알아두시면 좋습니다!
